🪟 🎨 Streams table design pass

tealjulia opened the pull request:
## What
addresses most of #17937
Design updates for the Connection Streams Table for Auto Detect Schema Changes.

![Screen Shot 2022-11-10 at 3 30 51 PM](https://user-images.githubusercontent.com/63751206/201199691-4e81a7c9-a2a9-4cbb-8fdb-76a57174ff04.png)

Figma prototypes used:
- sync mode dropdown based on [this prototype]( https://www.figma.com/file/liLQhcbpVHiEDW18kiXQe3/01_03_CONNECTIONS?node-id=972%3A36208)
- most of the rest of it based on [this prototype](https://www.figma.com/proto/liLQhcbpVHiEDW18kiXQe3/01_03_CONNECTIONS?page-id=857%3A24319&node-id=865%3A52984&viewport=124%2C-519%2C0.25&scaling=min-zoom&starting-point-node-id=865%3A52984&show-proto-sidebar=1)

Not addressed:
- not all of the background color changes match the prototype 100% (tracked separately in #19092)
- tooltip for the syncmode dropdown (tracked separately in #18243)
- table header alignment

## How
A lot of sass updates.  Not many changes beyond that aside from a wrapper component for `TextCell`s.

## Recommended reading order
Any.  But if you're looking at the page, the visual order (top to bottom) is approximately:
1. `StreamConnectionHeader`
2. `CatalogTreeBody`
3. `CatalogTreeTableHeader`
4. `CatalogTreeTableRow`
others sprinkled in 

## Note:
To check this out locally, make sure you have `REACT_APP_NEW_STREAMS_TABLE=true` in your `.env.development` file.

tealjulia commented on airbyte-webapp/src/components/connection/CatalogTree/next/StreamConnectionHeader.tsx:
```diff
@@ -16,12 +20,25 @@ export const StreamConnectionHeader: React.FC = () => {
-      <FontAwesomeIcon icon={faArrowRight} />
-      <div className={styles.connector}>{renderIcon(destinationDefinition.icon)} Destination</div>
+      <div className={sourceStyles}>
+        {renderIcon(sourceDefinition.icon)}{" "}
```
The icon styles predate this PR and match the values in Figma.  However, some logos appear _very_ small (Salesforce for one).  I'm in favor of adjusting it, but want to see if others agree before I go straying from the designs.

edmundito replied:
Looking at the preview, the logos look too small for me because of the irregular size. Either the width or width is consistently 15 px or the height but not both because it will make the icon shrink to fit both. Another difference with Figma is that the icons will not render as sharply as they do, so for example the Postgres icon doesn't look great there. We may need to revisit this with design to look acceptable in the implementation.

tealjulia replied:
agree

tealjulia commented on airbyte-webapp/src/components/connection/CatalogTree/next/CatalogTreeTableRow.tsx:
```diff
@@ -63,14 +63,13 @@ export const CatalogTreeTableRow: React.FC<StreamHeaderProps> = ({
     [styles.disabledChange]: changedSelected && !isStreamEnabled,
     [styles.selected]: isSelected,
     [styles.error]: hasError,
+    [styles.disabled]: !changedSelected && !isStreamEnabled,
```
Note that this makes the syncmode dropdown blend into the background.  I do see this in the designs as well, and users can't save changes to a disabled stream anyways currently... so maybe that's ok?  Since they can't change things, I wonder if we want to disable the dropdowns in these rows though.

edmundito replied:
I noticed that! I'm not a big fan, and maybe we can work around it by disabling it. Additionally, I feel the color contrast may be too low and we may need to revisit this with design.

edmundito commented on airbyte-webapp/src/components/connection/CatalogTree/next/SyncModeSelect.module.scss:
```diff
@@ -1,3 +1,6 @@
+
 .pillSelect {
-  width: 100%;
+  width: variables.$width-wide-menu;
```
100% did not work as expected?

tealjulia replied:
It did not.  Also, Nico pointed out that the SyncModeSelect should stay consistent width even when the other columns resize.

edmundito commented on airbyte-webapp/src/views/Connection/ConnectionForm/components/SyncCatalogField.module.scss:
```diff
@@ -7,4 +9,5 @@
   font-weight: 500;
   font-size: 14px;
   line-height: 17px;
+  padding: variables.$spacing-xl;
```
This gives it a huge padding to the top and bottom, maybe it's meant to be just the horizontal padding?

![Screen Shot 2022-11-09 at 09 39 44](https://user-images.githubusercontent.com/168664/200861064-3ab76b63-0a97-4926-bba4-1c2a65a22274.png)

edmundito commented on airbyte-webapp/src/components/connection/CatalogTree/next/StreamConnectionHeader.module.scss:
```diff
@@ -5,16 +5,35 @@ $icon-size: 15px;
-  height: $icon-size;
   width: $icon-size;
+  height: $icon-size;
+  padding-right: variables.$spacing-sm;
```
Could this be set as a `gap` property in the source/destination style?

edmundito commented on airbyte-webapp/src/components/connection/CatalogTree/CatalogTreeBody.module.scss:
```diff
@@ -1,7 +1,7 @@
   margin-bottom: 29px;
   max-height: 600px;
-  overflow-y: auto;
+  overflow-y: overlay;
```
Let's follow up on this and figure out what we can do.

tealjulia replied:
solved using `scrollbar-gutter`.  Still wonky on Safari if a user has an external mouse, but that seemed a reasonable tradeoff to Tim and I.

edmundito reviewed (commented):
Really great to see the table coming together! Followed up on some of your questions and pointed some things I noticed during testing.

edmundito commented:
Also, the error state is a bit different for this new table but currently, it follows what the current table does:

![Screen Shot 2022-11-09 at 09 38 34](https://user-images.githubusercontent.com/168664/200863996-f396d8cf-a719-47cc-be78-712f352355fb.png)

Instead, the primary key/cursor dropdowns should highlight red. This is something we could address as a separate issue because the dropdowns still need a lot of work. How would you like to handle this?

tealjulia commented:
> Also, the error state is a bit different for this new table but currently, it follows what the current table does:
> 
> ![Screen Shot 2022-11-09 at 09 38 34](https://user-images.githubusercontent.com/168664/200863996-f396d8cf-a719-47cc-be78-712f352355fb.png)
> 
> Instead, the primary key/cursor dropdowns should highlight red. This is something we could address as a separate issue because the dropdowns still need a lot of work. How would you like to handle this?

what do you think about rolling that into https://github.com/airbytehq/airbyte/issues/18243

tealjulia commented:
@edmundito addressed the comments you left above.  

Something in this PR caused us to lose the ellipses for cell contents. I got it to stop wrapping.  Will look later this afternoon at getting the ellipses back the way they're supposed to be...

edmundito reviewed (commented):
Did another pass. Code changes look good but noticed a few details while testing (Chrome)

1. Looks like the table headings are not aligning with the columns.
2. I noticed that the sync mode dropdown does not stretch to the full size of the column.
3. The heading arrow can be offset from the columns

Here's a screenshot pointing all 3
![Screen Shot 2022-11-10 at 09 30 32](https://user-images.githubusercontent.com/168664/201120616-8f66768b-8949-4a24-ade8-236f66632629.png)


There is a larger gap between the card heading and the search bar.

This is what's in the design:
![Screen Shot 2022-11-10 at 09 33 09](https://user-images.githubusercontent.com/168664/201120795-a850170f-d5c8-4596-944e-ab89683d9212.png)

And this is what I see:
![Screen Shot 2022-11-10 at 09 42 17](https://user-images.githubusercontent.com/168664/201121178-86cd422c-8eff-4521-81f7-b48b4a30d17e.png)

tealjulia commented:
> Did another pass. Code changes look good but noticed a few details while testing (Chrome)
> 
> 1. Looks like the table headings are not aligning with the columns.
> 2. I noticed that the sync mode dropdown does not stretch to the full size of the column.
> 3. The heading arrow can be offset from the columns
> 
> Here's a screenshot pointing all 3 ![Screen Shot 2022-11-10 at 09 30 32](https://user-images.githubusercontent.com/168664/201120616-8f66768b-8949-4a24-ade8-236f66632629.png)
> 
> There is a larger gap between the card heading and the search bar.
> 
> This is what's in the design: ![Screen Shot 2022-11-10 at 09 33 09](https://user-images.githubusercontent.com/168664/201120795-a850170f-d5c8-4596-944e-ab89683d9212.png)
> 
> And this is what I see: ![Screen Shot 2022-11-10 at 09 42 17](https://user-images.githubusercontent.com/168664/201121178-86cd422c-8eff-4521-81f7-b48b4a30d17e.png)



> 

I addressed item #2.  I propose that items 1 and 3 (heading alignment) be addressed in a separate PR.  I've updated the description to say that and removed the "closes" verbage so the Issue stays open.
